graph/telemetry/istio/appender/extensions.go

```diff
@@ -0,0 +1,360 @@
+			lCode, codeOk := m["status_code"]
+
+			if !protocolOk || !codeOk {
+				log.Warningf("Extension [%s] skipping %s, missing expected request TS labels", ext.Name, m.String())
+				continue
+			}
+
+			protocol = string(lProtocol)
+			code = string(lCode)
+
+			if skipRequestsGrpc && protocol == graph.GRPC.Name || skipRequestsHttp && protocol == graph.HTTP.Name {
+				continue
+			}
+		}
+
+		a.addTraffic(ext, trafficMap, metric, val, protocol, code, flags, secure, sourceCluster, sourceIsRoot, sourceNs, sourceName, destCluster, destNs, destName)
+	}
+}
+
+func (a ExtensionsAppender) addTraffic(ext config.ExtensionConfig, trafficMap graph.TrafficMap, metric string, val float64, protocol, code, flags, secure, sourceCluster, sourceIsRoot, sourceNs, sourceName, destCluster, destNs, destName string) {
+	isRoot := sourceIsRoot == "true"
+	source, _, err := a.addNode(ext, trafficMap, isRoot, sourceCluster, sourceNs, sourceName, a.GraphType)
+	if err != nil {
+		log.Warningf("Extension [%s] skipping extension addTraffic (source) in extension, %s", ext.Name, err)
+		return
+	}
+	dest, _, err := a.addNode(ext, trafficMap, false, destCluster, destNs, destName, a.GraphType)
+	if err != nil {
+		log.Warningf("Extension [%s] skipping extension addTraffic (dest), %s", ext.Name, err)
+		return
+	}
+
+	if isRoot {
+		url := a.getUrl(ext, source)
+		if url != urlNotFound {
+			source.Metadata[graph.IsExtension] = &graph.ExtInfo{
+				URL:  url,
+				Name: ext.Name,
+			}
+			dest.Metadata[graph.IsExtension] = &graph.ExtInfo{
+				URL:  url,
+				Name: ext.Name,
+			}
+		}
+	}
+
+	// Extensions may generate duplicate metrics by reporting from both the source and destination. To avoid
+	// processing the same information twice we keep track of the time series applied to a particular edge. The
+	// edgeTSHash incorporates information about the time series' source, destination and metric information,
+	// and uses that unique TS has to protect against applying the same information twice.
+	edgeTSHash := fmt.Sprintf("%x", md5.Sum([]byte(fmt.Sprintf("%s:%s:%s:%s:%s:%s", metric, source.Metadata[tsHash], dest.Metadata[tsHash], code, flags, destName))))
+
+	a.addEdgeTraffic(val, protocol, code, flags, secure, destName, source, dest, edgeTSHash)
+}
+
+func (a ExtensionsAppender) addNode(ext config.ExtensionConfig, trafficMap graph.TrafficMap, isRoot bool, cluster, namespace, name, graphType string) (*graph.Node, bool, error) {
+	id, nodeType, err := graph.Id(cluster, namespace, name, "", "", "", "", graphType)
+	if err != nil {
+		return nil, false, err
+	}
+	node, found := trafficMap[id]
+	if !found {
+		if isRoot {
+			// The supplied source information did not exactly match a node in the graph, see if we can make a good "guess"
+			node, found = a.findRootNode(trafficMap, cluster, namespace, name)
+			if !found && !a.ShowUnrooted {
+				return nil, false, fmt.Errorf("extension [%s] no root node found matching is_source_root=true, id=%s", ext.Name, id)
```

Comment:
should be `source_is_root`, no?

```suggestion
				return nil, false, fmt.Errorf("extension [%s] no root node found matching source_is_root=true, id=%s", ext.Name, id)
```